### PR TITLE
Feature/aeon request fix

### PR DESCRIPTION
**Pull request by sullivan1lsu**
Added code to handle Desktop with an interval to prevent timing issues. Added code to handle Mobile. Tested with the console.

### CODE REVIEW

File: our_files/custom.js [hunk 1]
```diff
@@ -1058,6 +1058,88 @@ var renameDueStatus = function() {
 };
 
 //Detail View Tasks -- ITEM_HOLD_LINK tasks
+/*
```
Comment (sullivan1lsu) [inline our_files/custom.js:1074]: Added comments here as a model: Purpose, URL to see in production, Python test that tests this functionality if it exists

File: our_files/custom.js [hunk 2]
```diff
@@ -1058,6 +1058,112 @@ var renameDueStatus = function() {
...
+            Unavailable
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+*/
+
+let aeonIntervalPlaceholder;
```
Comment (sullivan1lsu) [inline our_files/custom.js:1179]: Used for the interval timer;

File: our_files/custom.js [hunk 3]
```diff
@@ -1115,66 +1222,102 @@ var aeonRequest = function() {
...
-        .text();
-      var itemLocation = "&Location=" + encodeURIComponent(curLocation);
-      if (libr == SPEC_COLL || libr == ALT_SPEC_COLL) {
-        if (curLocation == REMOTE) {
-          requestType = "&Value=GenericRequestAllIronMountain";
-        } else {
-          requestType = "&Value=GenericRequestAll";
+  aeonIntervalPlaceholder = setInterval(function() {
+    jQuery(".detailItemsDiv .detailItemTable > tbody > tr.detailItemsTableRow")
+      .add(".detailItems .detailItemTable .detailChildRecord")
```
Comment (sullivan1lsu) [inline our_files/custom.js:1240]: Pulled in Mobile markup

File: our_files/custom.js [hunk 4]
```diff
@@ -1115,66 +1222,102 @@ var aeonRequest = function() {
...
+      .each(function() {
+        const mobileMOdifier = jQuery(this).hasClass("detailChildRecord")
```
Comment (sullivan1lsu) [inline our_files/custom.js:1242]: Doing this simplified the calls to those elements - mobile had an additional class for labels and values

File: our_files/custom.js [hunk 5]
```diff
@@ -1115,66 +1222,102 @@ var aeonRequest = function() {
...
+                '">' +
+                REQUEST_MATERIAL +
+                "</a></td>"
+            );
+
+            var destElem = $J(this)
+              .find(".detailItemsTable_SD_ITEM_HOLD_LINK")
+              .not(".hidden");
+            replaceItemHoldsElem(aeonElem, destElem);
+          } else if (jQuery(this).hasClass("detailChildRecord")) {
```
Comment (sullivan1lsu) [inline our_files/custom.js:1303]: Mobile markup.  Could be further simplified, but this works.

File: our_files/custom.js [hunk 6]
```diff
@@ -1115,66 +1222,102 @@ var aeonRequest = function() {
...
+            );
+
+            var destElem = $J(this)
+              .find(".asyncFieldSD_ITEM_HOLD_LINK")
+              .not(".hidden");
+
+            replaceItemHoldsElem(aeonElem, destElem);
+          }
+
+          clearInterval(aeonIntervalPlaceholder);
```
Comment (sullivan1lsu) [inline our_files/custom.js:1331]: Either flow, if we're here, clear the interval

**Comment by sullivan1lsu** [comment]
Mobile:

Before:
image

After:
image

I also confirmed the links are identical.

**Comment by davidcomeaux** [comment]
Wow! That was fast! Thank you so much!

File: our_files/custom.js [hunk 7]
```diff
@@ -668,11 +668,24 @@ Desktop Outgoing Markup:
...
+      </div>
+    </div>
+  </div>
+</div>
+
 Mobile Outgoing Markup: TBD
 */
 var renameItemHoldsColumn = function() {
-  $J('.detailItemTable_th:contains("Item Holds")').text("Request Item");
+  $J('.detailItemTable_th:contains("Item Holds")')
```
Comment (sullivan1lsu) [inline our_files/custom.js:686]: Rename column header on mobile

**Comment by GarrettArm** [comment]
These look great to me.  The code cleanup and the new functionality both

**Pull request merged**